src/inference/src/dev/make_tensor.cpp

```diff
@@ -359,16 +359,26 @@ class BaseRoiTensor {
     }
 
     void set_shape(ov::Shape new_shape) {
-        OPENVINO_ASSERT(new_shape.size() == m_shape.size());
-        for (auto new_dim = new_shape.cbegin(), max_dim = m_capacity.cbegin(); new_dim != new_shape.cend();
+        const auto last_new_dim = new_shape.crend();
```

Comment:
That in not a "last dim". It is the end of the vector
rename it please

Reply:
The last dim is used as it is common in algorithms to name end iterator as last